source/index.ts

```diff
@@ -184,6 +192,34 @@ class APIClient {
         return response.access_token;
     }
 
+    async _waitUploadSuccess(response: ItemResource, currentAttempt: number, maxRetries: number): Promise<ItemResource> {
+        if (!this._isUploadInProgress(response)) {
+            // We can return the response immediately if the upload is not in progress
+            return response;
+        }
+
+        if (currentAttempt >= maxRetries) {
+            throw new Error('Upload is still in progress after maximum retries');
+        }
+
+        // Wait for at least 5 seconds, with an exponential backoff, before checking again
+        const retryIn = retryInterval * (2 ** currentAttempt);
```

Comment:
The current values make 4 requests in the first minute, then:

- +1 minute
- +2 minutes
- +4 minutes

Exponent seems too high, GET requests are cheap

Reply:
Switched to a linear retry